core/src/main/java/bio/terra/pearl/core/dao/BaseJdbiDao.java

```diff
@@ -338,6 +358,39 @@ protected List<T> findAllByTwoProperties(String column1Name, List<?> column1Valu
         });
     }
 
+    protected List<T> findAllByThreeProperties(String column1Name, List<?> column1Values,
```

Comment:
Out of scope for this PR, but I do think we should consider using varargs for these types of DAO methods so we can handle an arbitrary number of properties, something like this (this code hasn't been thoroughly tested, I just had Copilot mock something up quickly):

```
public record ColumnProperty(String columnName, List<?> columnValues) {}

protected List<T> findAllByProperties(ColumnProperty... properties) {
    if (Arrays.stream(properties).map(p -> p.columnValues().size()).distinct().count() > 1) {
        throw new IllegalArgumentException("columnValues must all be the same size");
    }
    if (Arrays.stream(properties).anyMatch(p -> p.columnValues().isEmpty())) {
        // short circuit this case because bindList errors if list is empty
        return new ArrayList<>();
    }

    String whereClause = IntStream.range(0, properties[0].columnValues().size())
            .mapToObj(i -> Arrays.stream(properties)
                    .map(p -> "%s = :%sValue%d and ordering = %d".formatted(p.columnName(), p.columnName(), i, i))
                    .collect(Collectors.joining(" and ")))
            .collect(Collectors.joining(" or "));

    String sortValues = IntStream.range(0, properties[0].columnValues().size())
            .mapToObj(i -> Arrays.stream(properties)
                    .map(p -> "(:%sValue%d, %d)".formatted(p.columnName(), i, i))
                    .collect(Collectors.joining(", ")))
            .collect(Collectors.joining(", "));

    String joinClause = """
        join (values %s) as sortJoin %s
        on %s
        """.formatted(sortValues,
            Arrays.stream(properties)
                    .map(p -> "(%s, ordering)".formatted(p.columnName()))
                    .collect(Collectors.joining(", ")),
            Arrays.stream(properties)
                    .map(p -> "%s = sortJoin.%s".formatted(p.columnName(), p.columnName()))
                    .collect(Collectors.joining(" and ")));

    return jdbi.withHandle(handle -> {
        Query query = handle.createQuery("select * from %s %s where %s order by sortJoin.ordering"
                .formatted(tableName, joinClause, whereClause));
        for (int i = 0; i < properties[0].columnValues().size(); i++) {
            for (ColumnProperty p : properties) {
                query = query.bind("%sValue%d".formatted(p.columnName(), i), p.columnValues().get(i));
            }
        }
        return query.mapTo(clazz).list();
    });
}
```

Reply:
I like this idea a lot - I vote we make it a ticket to do. We could then do like Map.of/Map.ofEntries, and have helper methods like:

```
findAllByTwoProperties(String column1, Object val1, String column2, Object val2 ) {
     return findAllByProperties(new ColumnProperty(column1, val1), new ColumnProperty(column2, val2)
}
```

etc. so it's fully backwards compatible. Map.of has this up to 10 entries long, which could be cool to have that flexibility for our own DAO methods.